zone/client_packet.cpp

```diff
@@ -12806,87 +12806,101 @@ void Client::Handle_OP_SetTitle(const EQApplicationPacket *app)
+		return; 
+	}
+
+	if (GetClass() != WARRIOR){
 		return;
 	}
 
-	if (shield_target)
-	{
-		entity_list.MessageCloseString(
-			this, false, 100, 0,
-			END_SHIELDING, GetName(), shield_target->GetName());
-		for (int y = 0; y < 2; y++)
-		{
-			if (shield_target->shielder[y].shielder_id == GetID())
-			{
-				shield_target->shielder[y].shielder_id = 0;
-				shield_target->shielder[y].shielder_bonus = 0;
-			}
-		}
+	pTimerType timer = pTimerShieldAbility;
+
+	if (!p_timers.Expired(&database, timer, false)) {
+		uint32 remain = p_timers.GetRemainingTime(timer);
+		Message(Chat::White, "You can use the ability /shield in %d minutes %d seconds.", ((remain) / 60), (remain % 60));
+		return;
 	}
+
 	Shielding_Struct* shield = (Shielding_Struct*)app->pBuffer;
-	shield_target = entity_list.GetMob(shield->target_id);
-	bool ack = false;
-	EQ::ItemInstance* inst = GetInv().GetItem(EQ::invslot::slotSecondary);
-	if (!shield_target)
+
+	Mob* shield_target = entity_list.GetMob(shield->target_id);
+
+	if (!shield_target) {
 		return;
-	if (inst)
-	{
-		const EQ::ItemData* shield = inst->GetItem();
-		if (shield && shield->ItemType == EQ::item::ItemTypeShield)
-		{
-			for (int x = 0; x < 2; x++)
-			{
-				if (shield_target->shielder[x].shielder_id == 0)
-				{
-					entity_list.MessageCloseString(
-						this, false, 100, 0,
-						START_SHIELDING, GetName(), shield_target->GetName());
-					shield_target->shielder[x].shielder_id = GetID();
-					int shieldbonus = shield->AC * 2;
-					switch (GetAA(197))
-					{
-					case 1:
-						shieldbonus = shieldbonus * 115 / 100;
-						break;
-					case 2:
-						shieldbonus = shieldbonus * 125 / 100;
-						break;
-					case 3:
-						shieldbonus = shieldbonus * 150 / 100;
-						break;
-					}
-					shield_target->shielder[x].shielder_bonus = shieldbonus;
-					shield_timer.Start();
-					ack = true;
-					break;
-				}
-			}
-		}
-		else
-		{
-			Message(0, "You must have a shield equipped to shield a target!");
-			shield_target = 0;
-			return;
-		}
 	}
-	else
-	{
-		Message(0, "You must have a shield equipped to shield a target!");
-		shield_target = 0;
+	
+	if (shield_target->IsNPC()) {
+		MessageString(Chat::White, SHIELD_TARGET_NPC);
 		return;
 	}
-	if (!ack)
-	{
+
+	if (shield_target->GetID() == GetID()) { //Client will give message "You can not shield yourself"
+		return;
+	}
+	   
+	//You are a 'Shield Target' already have a 'Shielder'
+	if (GetShielderID() || shield_target->GetShielderID()) {
 		MessageString(Chat::White, ALREADY_SHIELDED);
-		shield_target = 0;
 		return;
 	}
+
+	//You are being shielded or already have a 'Shield Target'
+	if (GetShieldTargetID() || shield_target->GetShieldTargetID()) {
+		MessageString(Chat::White, ALREADY_SHIELDING);
+		return;
+	}
+	
+	//AA to increase SPA 230 extended shielding
+	int max_shlder_distance = 15;
+	max_shlder_distance += aabonuses.ExtendedShielding + itembonuses.ExtendedShielding + spellbonuses.ExtendedShielding;
```

Comment:
Can we full word the variables here as well for readability `max_shielder_distance`